src/execution_plan/optimizations/traverse_order_utils.c

```diff
@@ -17,10 +17,6 @@ static bool _AlgebraicExpression_IsVarLen
 	ASSERT(qg  != NULL);
 	ASSERT(exp != NULL);
 
-	// variable length expression contains a single edge operand
-	if(AlgebraicExpression_OperandCount(exp) != 1  ||
-	   AlgebraicExpression_OperationCount(exp, AL_EXP_ALL) > 0) return false;
-
```

Comment:
Can you please explain why was this condition removed ?

Reply:
Because we can validly have transpose and addition ops here, and in the case of addition we may have multiple operands. The following query causes a transposed traversal of one operand:
```
MATCH (b)<-[*]-(a:L {v: 5})<--(a) WHERE b.v = 10 RETURN a
```
And this bidirectional traversal will create `E + T(E)`:
```
 MATCH (a:L {v: 5})-[E*]-(b:L) WHERE a <> b RETURN b
```
Traversing multiple label types would also build an addition of N operands.

Reply:
👍 thank you!